src/graphql/typeDefs/User/type.js

```diff
@@ -1,7 +1,92 @@
+import InputTypes from './input';
+
 export default `
+
+type BaseFile {
+  key: String!
+  head: String
+  tail: String
+  path: String
+  history: [String]!,
+  name: String!
+  ext: String!
+  contents: String!
+  source: String
+}
+
+type CompletedChallenge {
+  id: String
+  files: File
+  bucket: String
+  completedDate: Int
+  numOfAttempts: Int
+}
+
+type File {
+  indexcss: BaseFile
+  indexhtml: BaseFile
+  indexjs: BaseFile
+  indexjsx: BaseFile
+}
+
+type PortfolioItem {
+  description: String
+  image: String
+  title: String
+  url: String
+}
+
+type Progress {
+  calendar: [Int!]!
+  streak: Streak!
+}
+
+type ProgressTimestamp {
+  timestamp: Int!
+  completedChallenge: String
+}
+
+type Streak {
+  longest: Int!
+  current: Int!
+}
+
 type User {
-  _id: ID @isAuthenticatedOnField
-  email: String
+  email: String!
+  progressTimestamps: [ProgressTimestamp]!,
+  isBanned: Boolean
+  isCheater: Boolean
+  githubURL: String
```

Comment:
`githubURL` => `github` ? like other socials?

Reply:
Not required any more